addons/sale_project/tests/test_sale_project.py

```diff
@@ -499,3 +500,40 @@ def test_project_creation_on_so_confirm_with_default_plan_with_company_in_settin
         self.assertEqual(sale_order.analytic_account_id.company_id, analytic_plan.company_id, "The company_id of the account created should be the company of the default analytic plan of the setting.")
         self.assertEqual(sale_order.analytic_account_id.plan_id, analytic_plan, "The plan of the account created should be the default analytic plan of the setting")
         self.assertEqual(sale_order.analytic_account_id, sale_order.project_ids.analytic_account_id, "The project created for the SO and the SO should have the same account.")
+
+    def test_allow_editing_tasks_sale_line_id(self):
+        """
+        Ensure that when selected task have same partner than only sale_order_line can be edit otherwise
+        if selected task have diffrent partner or no partner than raise validation error and sale_line_id can not edit
+        """
+        deco_partner = self.env['res.partner'].create({'name': "Deco adict"})
+        sale_order = self.env['sale.order'].create([
+            {
+                'partner_id': self.partner.id,
+                'order_line': [
+                    Command.create({'product_id': self.product_order_service3.id}),
+                ]
+            },
+        ])
+
+        sale_order.action_confirm()
+        sale_order_project_id = sale_order.project_id.id
+        sale_order_task = sale_order.tasks_ids
+
+        # create to select diffrent task and check partner of selected task
+        tasks = self.env['project.task'].create([
+            {'name': 'Test Task 1', 'project_id': sale_order_project_id, 'partner_id': self.partner.id},
+            {'name': 'Test Task 2', 'project_id': sale_order_project_id, 'partner_id': self.partner.id},
+        ])
```

Comment:
```suggestion
        tasks = self.env['project.task'].create([{
            'name': 'Test Task',
            'project_id': sale_order_project_id,
            'partner_id': self.partner.id
        } for dummy in range(2)])
```
We could do this :wink: